MenuManager.cpp

```diff
@@ -5,43 +5,133 @@
 
 using namespace std;
 
-void MenuManager::getUserInput() {
+int MenuManager::getUserInput() {
+	cin.clear();
+	int input;
 	cout << "Please enter the numerical digit of the option you would like to select...\n";
-	cout << "Enter here: "; cin >> userInput;
+	cout << "Enter here: "; cin >> input;
 	cout << endl;
+	return input;
+}
+
+void MenuManager::startMenu() {
+	while (!quit) {
+		showWelcomeMenu();
+	}
 }
 
 void MenuManager::showWelcomeMenu() {
+	cout << "Welcome" << endl;
 	cout << "--------------------------------------------------------------------------\n";
 	cout << "Welcome to the Library Management System!\n\n";
 	cout << setw(5) << "" << "1 - Search Books\n";
 	cout << setw(5) << "" << "2 - Sign In\n";
 	cout << setw(5) << "" << "3 - Register\n";
 	cout << setw(5) << "" << "4 - Staff Sign In\n";
 	cout << setw(5) << "" << "5 - Exit\n\n";
+	processWelcomeMenuInput(getUserInput());
 }
 
-void MenuManager::processWelcomeMenuInput() {
+void MenuManager::processWelcomeMenuInput(int userInput) {
 	system("cls");
 	switch (userInput) {
 		case 1:
 			cout << "Feature unavailable. Please try again later." << endl;
 			break;
 		case 2:
-			cout << "Feature unavailable. Please try again later." << endl;
+			showMemberMainMenu();
 			break;
 		case 3:
 			cout << "Feature unavailable. Please try again later." << endl;
 			break;
 		case 4:
-			cout << "Feature unavailable. Please try again later." << endl;
+			showAdminMainMenu();
 			break;
 		case 5:
 			cout << "Exiting..." << endl;
+			quit = true;
 			break;
 		default:
 			cout << "Invalid option selected...\n";
 			break;
 
 	}
+}
+
+void MenuManager::showMemberMainMenu() {
+	cout << "Welcome \"Username\"\n\n";
+	cout << "Main Menu" << endl;
+	cout << "--------------------------------------------------------------------------\n";
+	cout << setw(5) << "" << "1 - Search Books\n";
+	cout << setw(5) << "" << "2 - View My Loans\n";
+	cout << setw(5) << "" << "3 - Edit My Account Info\n";
+	cout << setw(5) << "" << "4 - Sign Out\n";
+	cout << setw(5) << "" << "5 - Exit\n\n";
+	processMemberMainMenu(getUserInput());
+}
+
+void MenuManager::processMemberMainMenu(int userInput) {
+	system("cls");
+	switch (userInput)
+	{
+	case 1:
+		cout << "Feature unavailable. Please try again later." << endl;
+		break;
+	case 2:
+		cout << "Feature unavailable. Please try again later." << endl;
+		break;
+	case 3:
+		cout << "Feature unavailable. Please try again later." << endl;
+		break;
+	case 4:
+		break;
+	case 5:
+		cout << "Exiting...";
+		quit = true;
+		break;
+	default:
+		cout << "Invalid option selected...\n";
+		break;
+	}
+}
+
+void MenuManager::showAdminMainMenu() {
+	cout << "Welcome \"Username\"\n\n";
+	cout << "Administrator Menu" << endl;
+	cout << "--------------------------------------------------------------------------\n";
+	cout << setw(5) << "" << "1 - Search Books\n";
+	cout << setw(5) << "" << "2 - Add/Edit Books\n";
```

Comment:
Adding and editing books should be separate features, otherwise another menu will be necessary to select between adding and editing a book